### ISSUE TITLE
Failure trying to use Zinc in WSL2 Windows-Linux Filesystem Mount

### ISSUE THREAD

**Issue by lihaoyi** [sbt/zinc#752]
This failure occurs because the filesystem mount that allows the WSL2 environment to see Windows files does not appear to allow setting timestamps using the POSIX API that you guys are using. Is there some way to workaround this?

```
$ pwd
/mnt/c

$ curl -L https://www.lihaoyi.com/mill/example-3.zip > example-3.zip

$ unzip example-3.zip

$ cd example-3

$ ./mill -i foo.compile
[25/25] foo.compile
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.protobuf.UnsafeUtil (file:/home/ubuntu/.cache/coursier/v1/https/repo1.maven.org/maven2/com/google/protobuf/protobuf-java/3.3.1/protobuf-java-3.3.1.jar) to field java.nio.Buffer.address
WARNING: Please consider reporting this to the maintainers of com.google.protobuf.UnsafeUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
1 targets failed
foo.compile java.io.IOException: Operation not permitted, file: /mnt/c/test/example-3/out/foo/compile/dest/zinc
    sbt.internal.io.MilliPosixBase.checkedIO(Milli.scala:99)
    sbt.internal.io.PosixMilliLongUtim.setModifiedTimeNative(Milli.scala:186)
    sbt.internal.io.PosixMilliLongUtim.setModifiedTimeNative(Milli.scala:180)
    sbt.internal.io.MilliNative.setModifiedTime(Milli.scala:68)
    sbt.internal.io.Milli$.setModifiedTime(Milli.scala:364)
    sbt.io.IO$.setModifiedTimeOrFalse(IO.scala:1236)
    sbt.io.IO$.copyLastModified(IO.scala:1263)
    sbt.io.IO$.copyFile(IO.scala:731)
    sbt.io.IO$.move(IO.scala:927)
    sbt.internal.inc.FileAnalysisStore$BinaryFileStore.set(FileAnalysisStore.scala:95)
    mill.scalalib.worker.ZincWorkerImpl.compileInternal(ZincWorkerImpl.scala:439)
    mill.scalalib.worker.ZincWorkerImpl.$anonfun$compileMixed0$1(ZincWorkerImpl.scala:270)
    mill.api.KeyedLockedCache$RandomBoundedCache.withCachedValue(KeyedLockedCache.scala:31)
    mill.scalalib.worker.ZincWorkerImpl.withCompilers(ZincWorkerImpl.scala:341)
    mill.scalalib.worker.ZincWorkerImpl.compileMixed0(ZincWorkerImpl.scala:269)
    mill.scalalib.worker.ZincWorkerImpl.compileMixed(ZincWorkerImpl.scala:239)
    mill.scalalib.ScalaModule.$anonfun$compile$2(ScalaModule.scala:143)
    mill.define.ApplyerGenerated.$anonfun$zipMap$9(ApplicativeGenerated.scala:21)
    mill.define.Task$MappedDest.evaluate(Task.scala:379)

**Comment by lihaoyi** [comment]
@eed3si9n have you seen this issue before? It's not clear to me why we're using a platform-specific `setModifiedTimeNative`; would a workaround be to try-catch and fall back to the JDK-provided version if it fails?

**Comment by SethTisue** [comment]
history in sbt/io@3c06245, sbt/io@867da51

**Comment by lihaoyi** [comment]
I cannot reproduce this on the latest version of Zinc, so I'm closing this assuming it was fixed at some point between 1.2.5 and 1.4.0-M5
